xtask/src/main.rs

```diff
@@ -0,0 +1,69 @@
+use std::{env, process::Command};
+
+type DynError = Box<dyn std::error::Error>;
+
+fn main() {
+    if let Err(e) = execute_task() {
+        eprintln!("{e}");
+        std::process::exit(-1);
+    }
+}
+
+fn execute_task() -> Result<(), DynError> {
+    let task = env::args().nth(1);
+    match task.as_deref() {
+        Some("install-tools") => install_tools()?,
+        _ => {
+            return Err(Box::from(get_help_string(task.as_deref())));
+        }
+    }
+    Ok(())
+}
+
+fn install_tools() -> Result<(), DynError> {
+    println!("Installing project tools...");
+
+    let install_args: Vec<Vec<&str>> = vec![
```

Comment:
I would drop the explicit type annotation, it's probably not necessary and all the `vec![...]` calls make it fairly easy to see at a glance.

Reply:
Have pushed a new PR (https://github.com/google/comprehensive-rust/pull/2707) for this and the other suggestions.